app/controllers/dashboard_controller.rb

```diff
@@ -552,8 +552,10 @@ def authenticate
     when :wait_for_task
       # noop, page content already set by initiate_wait_for_task
     when :pass
+      miq_api_token = require_api_token ? generate_ui_api_token(user[:name]) : nil
       render :update do |page|
         page << javascript_prologue
+        page << "sessionStorage.miq_token = '#{miq_api_token}'" if miq_api_token
```

Comment:
Should probably be `page << "sessionStorage.miq_token = '#{j_str miq_api_token}';" if miq_api_token` - `j_str` to prevent "bad" characters breaking the rest of the script, and that extra semicolon just to be sure the statement really ends here (for example, if the next line started with a `+` sign..).